### ISSUE TITLE
Godot connecting internet without asking me (Even if the firewall is turned on)

### ISSUE THREAD

**Issue by icnikerazah** [godotengine/godot#13936]
**macOs Sierra**

**Issue description:**
Godot connecting internet without asking me (Even if the firewall is turned on)
there is no Connect/Disconnect options in godot (Offline Editor)

**Steps to reproduce:**
1.Godot blocked in Firewall
2.Godot is active in Network Activity Monitor(macos sierra)

**Link to minimal example project:**

**Comment by reduz** [comment]
I noticed this too, I wonder what's going on..

**Comment by vnen** [comment]
Is this happening when idling or when previewing the game? Because the debugging happens over network protocol, so it might register network activity.

Also: you didn't mention the Godot version.

**Comment by icnikerazah** [comment]
godot version -> http://godot3builds.digitecnology.com/builds/2017-12-02.13:00:36.5ddf288c7/godot-v3.0-osx.fat.zip
This is happening when AssetLib activity, maybe also debug i dont know. Godot bypassing sierra firewall. I want to use editor offline,(security) i am a paranoid.

**Comment by Zireael07** [comment]
AssetLib is stored externally, so when you use the assetlib you connect to the internet.

**Comment by reduz** [comment]
but we made it so unless you open the assetlib, it wont connect

**Comment by icnikerazah** [comment]
"Zireael07 : AssetLib is stored externally, so when you use the assetlib you connect to the internet."

In this case Assetlib becomes a trojan! godot is a virus.

**Comment by akien-mga** [comment]
Well feel free not to use the AssetLib.

**Comment by reduz** [comment]
@icnikerazah We took special care with the assetlib, if you dont use it it won't connect.

**Comment by icnikerazah** [comment]
How can I be sure that Gdot can not copy harmful files to my computer? My firewall is ineffective.

**Comment by reduz** [comment]
@icnikerazah you have the full source code and there are thousands of users watching, I would think you are pretty safe. I mean, this is not a conspiracy to steal your files, we are making a game engine here.

**Comment by icnikerazah** [comment]
i think godot should respect peoples's OS Firewall. This is not a professional approach. All Other Apps respect user preferences.

**Comment by bruvzg** [comment]
@icnikerazah what do you mean by "respect firewall"? I don't think macOS build in firewall is capable of blocking outgoing connections at all (at least there is no UI for it).

**Comment by icnikerazah** [comment]
You are right, i was wrong. i am not a network expert. outgoing connections are safe.

**Comment by bruvzg** [comment]
@icnikerazah Godot uses incoming connection for debugging. Debugger is listening on port 6007 to accept connections form running projects.

**Comment by bruvzg** [comment]
Same applies to Windows default firewall as well.

By default debugger is listening on all interfaces, while this is not a security issue by itself, it is probably redundant, and firewall warnings may be frustrating to users.

This can be "fixed" by only listening to `remote_host` specified in editor settings (default to localhost, and shouldn't trigger firewall), but I'm not sure if this won't cause any problems with remote debugging.

May be `remote_host` should be changed to check list with all interfaces, with only localhost selected by default.

https://github.com/godotengine/godot/blob/master/editor/script_editor_debugger.cpp#L1156

```c++
String current = EditorSettings::get_singleton()->has_setting("network/debug/remote_host") ? EditorSettings::get_singleton()->get("network/debug/remote_host") : "*";
int remote_port = (int)EditorSettings::get_singleton()->get("network/debug/remote_port");
if (server->listen(remote_port, IP_Address(current)) != OK) {
	EditorNode::get_log()->add_message(String("Error listening on port ") + itos(remote_port), true);
	return;
}
```

**Comment by Faless** [comment]
> May be remote_host should be changed to check list with all interfaces, with only localhost selected by default.

By default the editor listen on `127.0.0.1` i.e. localhost. You can change that setting in the Editor Settings (where you have the list of all interface), but the default is `127.0.0.1`

**Comment by bruvzg** [comment]
@Faless `remote_host` editor setting is passed to running project, editor always listen on all interfaces.

https://github.com/godotengine/godot/blob/master/editor/script_editor_debugger.cpp#L1156 `listen(uint16_t p_port, const IP_Address &p_bind_address = IP_Address("*"))` is called without address specified.

**Comment by Faless** [comment]
@bruvzg, you are right, my bad! I must have confused it with the client setting.

So I guess we could use the same option used for by the client to connect when run from the editor (which defaults to 127.0.0.1 and guarantee that the debugger client and server will always talk the same language).
https://github.com/godotengine/godot/blob/master/editor/editor_run.cpp#L44

And maybe add an additional boolean settings `Listen on all interfaces`.

Thoughts?
